### PR TITLE
[gui-tests][full-ci] added tests for adding folder to sync

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
### Description
Added test for adding a folder in remote destination to sync.

### Related issue
- https://github.com/owncloud/client/issues/11653

### CODE REVIEW

File: test/gui/shared/scripts/helpers/SetupClientHelper.py [hunk 1]
```diff
@@ -127,6 +127,7 @@ def setUpClient(username, displayName, space="Personal"):
     0/url={local_server}
     0/user={displayUserFirstName}
     0/version=1
+    0\supportsSpaces={supportsSpaces}
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/SetupClientHelper.py:130]: :+1:

File: test/gui/shared/scripts/pageObjects/SyncConnection.py [hunk 2]
```diff
@@ -29,6 +29,49 @@ class SyncConnection:
...
+        "name": "addFolderButton",
+        "type": "QPushButton",
+        "visible": 1,
+    }
+    ADD_FOLDER_INPUT = {
+        "buddy": names.create_Remote_Folder_Enter_the_name_of_the_new_folder_to_be_created_below_QLabel,
+        "type": "QLineEdit",
+        "unnamed": 1,
+        "visible": 1,
+    }
```
Comment (saw-jan) [inline test/gui/shared/scripts/pageObjects/SyncConnection.py:55]: AFAIK, we should already have this selector. Also, the best place for all these should be `SyncConnectionWizard`

File: test/gui/tst_syncing/test.feature [hunk 3]
```diff
@@ -480,3 +480,21 @@ Feature: Syncing files
         Then the folder "simple-folder" should not exist on the file system
         And the folder "test-folder/sub-folder2" should exist on the file system
         And the folder "test-folder/sub-folder1" should not exist on the file system
+
+
+    Scenario: Add folder to sync
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:485]: Maybe we can integrate these to existing scenario (add folder sync connection) if possible

File: test/gui/shared/scripts/names.py [hunk 4]
```diff
@@ -49,3 +49,6 @@
 o_folderList_Personal_QModelIndex = {"column": 0, "container": stack_folderList_QTreeView, "text": "Personal", "type": "QModelIndex"}
 add_Folder_Sync_Connection_tableView_QTableView = {"name": "tableView","type": "QTableView","visible": 1,"window": add_Folder_Sync_Connection_OCC_FolderWizard}
 stack_scrollArea_QScrollArea = {"container": settings_stack_QStackedWidget, "name": "scrollArea", "type": "QScrollArea", "visible": 1}
+create_Remote_Folder_QInputDialog = {"type": "QInputDialog", "unnamed": 1, "visible": 1, "windowTitle": "Create Remote Folder"}
+create_Remote_Folder_Enter_the_name_of_the_new_folder_to_be_created_below_QLabel = {"text": "Enter the name of the new folder to be created below '/':", "type": "QLabel", "unnamed": 1, "visible": 1, "window": create_Remote_Folder_QInputDialog}
+groupBox_folderTreeWidget_QTreeWidget = {"container": add_Folder_Sync_Connection_groupBox_QGroupBox, "name": "folderTreeWidget", "type": "QTreeWidget", "visible": 1}
```
Comment (saw-jan) [inline test/gui/shared/scripts/names.py:54]: Please, check if we need these

File: test/gui/shared/steps/sync_context.py [hunk 5]
```diff
@@ -237,3 +237,42 @@ def step(context, action):
...
+@When('the user refreshes the remote destination in the sync connection wizard')
+def step(context):
+    SyncConnectionWizard.refresh_remote()
+
+
+@Then(
+    r'the folder "([^"]*)" should be present and (selected|not selected) in the remote destination wizard',
+    regexp=True,
+)
+def step(context, folder_name, is_selected):
```
Comment (saw-jan) [inline test/gui/shared/steps/sync_context.py:266]:
```suggestion
def step(context, folder_name, selected):
```

File: test/gui/shared/steps/sync_context.py [hunk 6]
```diff
@@ -237,3 +237,42 @@ def step(context, action):
...
+    has_folder, folder_selector = SyncConnectionWizard.has_remote_folder(folder_name)
+    if is_selected == 'selected':
+        expected = True
+    else:
+        expected = False
```
Comment (saw-jan) [inline test/gui/shared/steps/sync_context.py:271]:
```suggestion
```

File: test/gui/shared/steps/sync_context.py [hunk 7]
```diff
@@ -237,3 +237,42 @@ def step(context, action):
...
+
+    test.compare(True, has_folder, "Folder should be in the remote list")
+    test.compare(
+        expected,
```
Comment (saw-jan) [inline test/gui/shared/steps/sync_context.py:275]:
```suggestion
    test.compare(
        selected == 'selected',
```

File: test/gui/shared/scripts/pageObjects/SyncConnectionWizard.py [hunk 8]
```diff
@@ -251,3 +293,61 @@ def syncSpace(spaceName):
...
+            )
+            try:
+                if parent_container:
+                    squish.doubleClick(parent_container)
+
+                squish.waitForObject(folder_selector)
+
+                parent_container = folder_selector
+            except Exception as e:
+                print(e)
```
Comment (saw-jan) [inline test/gui/shared/scripts/pageObjects/SyncConnectionWizard.py:347]:
```suggestion
            except:
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: test/gui/shared/scripts/names.py (hunk 4 above)
Comment (PrajwolAmatya) [reply test/gui/shared/scripts/names.py:54]: Yes, these selectors are being used in the test.

**Review by saw-jan — APPROVED** [review]

**Pull request merged**